### PR TITLE
Fix network agg resolution

### PR DESCRIPTION
**Pull request by ciaranschutte**
Fixes bucket resolution 

example: 
Server1 buckets: [Cat, Dog]
Server2 buckets: [Cat, Bird]
Output: [Cat, Dog, Bird]

### CODE REVIEW

File: modules/server/src/network/typeDefs/networkAggregations.ts [hunk 1]
```diff
@@ -1,13 +1,28 @@
...
 // TODO: Placeholder to expand type
 const networkAggregations = new GraphQLObjectType({
 	name: 'NetworkAggregations',
 	fields: {
 		bucket_count: {
 			type: GraphQLInt,
 		},
+		buckets: {
+			type: new GraphQLList(bucket),
+		},
```
Comment (ciaranschutte) [inline modules/server/src/network/typeDefs/networkAggregations.ts:25]: extend GQL typings for GQL query response

File: modules/server/src/network/typeDefs/aggregations.ts [hunk 2]
```diff
@@ -2,11 +2,26 @@ import { GraphQLObjectType, GraphQLSchema } from 'graphql';
...
+ */
+const convertToGQLObjectType = (networkFieldTypes) => {
+	return networkFieldTypes.reduce((allFields, currentField) => {
+		const field = {
+			[currentField.name]: { type: singleToNetworkAggregationMap.get(currentField.type) },
+		};
+		return { ...allFields, ...field };
+	}, {});
+};
+
```
Comment (ciaranschutte) [inline modules/server/src/network/typeDefs/aggregations.ts:19]: isolate function

File: modules/server/src/network/aggregations/index.ts [hunk 3]
```diff
@@ -44,21 +44,46 @@ export const resolveToNetworkAggregation = (
...
+				if (existingBucket) {
+					// update existing bucket
+					computedBuckets[existingBucketIndex] = {
+						...existingBucket,
+						doc_count: existingBucket.doc_count + doc_count,
+					};
+				}
+			} else {
+				computedBuckets.push(bucket);
+			}
```
Comment (ciaranschutte) [inline modules/server/src/network/aggregations/index.ts:82]: Size of buckets can be large
Using `forEach` to either update "found bucket" or add to resolved buckets

Avoids using functional operations on the entire "resolved buckets" and then having a large array of Buckets and needing to flatten and filter on same Bucket `key`

No way around searching buckets for existing bucket unfortunately as the data structure is an array and a key lookup is not an option.

**Comment by ciaranschutte** [comment]
CI failing is known issue

File: modules/server/src/network/aggregations/index.ts (hunk 3 above)
Comment (justincorrigible) [reply modules/server/src/network/aggregations/index.ts:82]: the lookup has always been a pain point... perhaps we should make a ticket for future improvement to make the buckets a dictionary available internally in the Arranger server for this and other lookups, even if it's ultimately turned in the UI as an array for display purposes.

File: modules/server/src/network/aggregations/index.ts [hunk 4]
```diff
@@ -44,21 +44,46 @@ export const resolveToNetworkAggregation = (
...
+				const existingBucket = computedBuckets[existingBucketIndex];
+				if (existingBucket) {
+					// update existing bucket
+					computedBuckets[existingBucketIndex] = {
+						...existingBucket,
+						doc_count: existingBucket.doc_count + doc_count,
+					};
+				}
+			} else {
+				computedBuckets.push(bucket);
```
Comment (justincorrigible) [inline modules/server/src/network/aggregations/index.ts:81]: lol I don't think I'll ever be happy/comfortable with mutational logic.

File: modules/server/src/network/aggregations/index.ts [hunk 5]
```diff
@@ -44,21 +44,46 @@ export const resolveToNetworkAggregation = (
  * Takes an array of the same aggregation type and computes the singular type
  * eg. NumericAggregation => NetworkNumericAggregation
  *
+ * Note for operations on Buckets -  size can be large
```
Comment (justincorrigible) [inline modules/server/src/network/aggregations/index.ts:47]: "size" alone is ambiguous, in that a large bucket doesn't add complexity, whereas the count of buckets is what I think you're trying to highlight here
```suggestion
 * Note for operations on Buckets -  the size of the array can be large (e.g. total bucket count), complicating lookups, etc.
```

File: modules/server/src/network/aggregations/index.ts (hunk 3 above)
Comment (ciaranschutte) [reply modules/server/src/network/aggregations/index.ts:82]: https://github.com/overture-stack/arranger/issues/886

**Pull request merged**